objects/src/transaction/transaction_id.rs

```diff
@@ -89,7 +91,7 @@ impl From<&ExecutedTransaction> for TransactionId {
         let input_notes_hash = tx.input_notes().commitment();
         let output_notes_hash = tx.output_notes().commitment();
         Self::new(
-            tx.initial_account().hash(),
+            tx.initial_account().proof_init_hash(),
```

Comment:
I guess this was an error before - right? I'm thinking we should create a test to make sure that transaction ID remains the same as a transaction moves from executed to proven (or maybe we can integrate into existing tests?).

Reply:
No, this is not a bugfix, just because of changing in input argument it was more convenient to use `proof_init_hash`. If it's confusing, it might be better to revert to account's hash usage here and implement additional logic for handling new accounts here.

Reply:
@bobbinth I'll implement this test and will fix the bug, thank you!

Reply:
> Just to clarify: I think the bug is already fixed - we just need to confirm this via a test.

Got it!

Reply:
@bobbinth, I've finally added checking into `prove_and_verify_transaction` function, which is used in many integration tests, including tests for minting new accounts in faucet. Please, look into it, if you have a chance, I think, it will be enough for checking this bug.

Reply:
Thank you! To confirm: if we just updated `prove_and_verify_transaction()` function and kept everything else the same, would there be any failing tests? (if so, which ones?).

Reply:
@bobbinth no, actually nothing failed in `next` branch after adding this check into `prove_and_verify_transaction` function.

Reply:
hmmm - are we sure then that we are testing executing transactions against new accounts? Is there any test that you see that does that?

(it is of course possible that I'm wrong and there was no bug to begin with, but I'd like to be sure).

Reply:
I tried to analyze our tests and now I can say we don't have any tests in miden-base which produce new account in transaction.

Reply:
Could you add one such test? This could be something like consuming a note for a new account.

You could do it in a different PR based on this branch, and then, once it is done, we can merge both these PRs.

Reply:
@bobbinth, I created an issue for implementing of this test: https://github.com/0xPolygonMiden/miden-base/pull/529#discussion_r1524036222
Can we merge this PR then? You would probably need to do a final look.